weaviate/collections/filters.py

```diff
@@ -2,40 +2,50 @@
 from typing import Any, Dict, List, Literal, Optional, cast, overload
 
 from weaviate.collections.classes.filters import (
+    _FilterValue2,
+    _MultiTargetRef,
+    _SingleTargetRef,
     _Filters,
     _FilterAnd,
     _FilterOr,
     _FilterValue,
     _GeoCoordinateFilter,
     FilterValues,
+    _FilterTargets,
 )
-from weaviate.util import _datetime_to_string
+from weaviate.util import _ServerVersion, _datetime_to_string
 from weaviate.types import TIME
 from weaviate.proto.v1 import search_get_pb2
 
 
 class _FilterToGRPC:
     @overload
     @staticmethod
-    def convert(weav_filter: Literal[None]) -> None:
+    def convert(weav_filter: Literal[None], weaviate_version: _ServerVersion) -> None:
         ...
 
     @overload
     @staticmethod
-    def convert(weav_filter: _Filters) -> search_get_pb2.Filters:
+    def convert(weav_filter: _Filters, weaviate_version: _ServerVersion) -> search_get_pb2.Filters:
         ...
 
     @staticmethod
-    def convert(weav_filter: Optional[_Filters]) -> Optional[search_get_pb2.Filters]:
+    def convert(
+        weav_filter: Optional[_Filters], weaviate_version: _ServerVersion
+    ) -> Optional[search_get_pb2.Filters]:
         if weav_filter is None:
             return None
         if isinstance(weav_filter, _FilterValue):
+            return _FilterToGRPC.__value_filter_old(weav_filter)
+        elif isinstance(weav_filter, _FilterValue2):
+            if weaviate_version < _ServerVersion(major=1, minor=23, patch=0):
```

Comment:
I would use the `is_at_least` method on the `_ServerVersion` class here. In hindsight, it should be used everywhere but I recall not using in tests so will change in a different PR

So `if not weaviate_version.is_at_least(1, 23, 0)`